### PR TITLE
544 duplicate controlled field values

### LINKED ISSUE #544: (5)Duplicate results for controlled fields with inheritance
Duplicate controlled field values are displayed with work->work and collection->work inheritance. Here is an example on sandbox.
http://digitalcollections-staging-sandbox.library.ucsc.edu/concern/works/m326m177s?locale=en

### PR DESCRIPTION
**Pull request by snehagunduraoUL**
resolves #544

### CODE REVIEW

File: app/indexers/work_indexer.rb [hunk 1]
```diff
@@ -104,17 +104,17 @@ def inherit_fields solr_doc
...
-        if (solr_doc[field.solr_name].present? && ScoobySnacks::METADATA_SCHEMA.add_parent_value_display_field_names.include?(field.name))
+        if (solr_doc[field.solr_name]!= parent_doc[field.solr_name] && ScoobySnacks::METADATA_SCHEMA.add_parent_value_display_field_names.include?(field.name))
           solr_doc[field.solr_name].push(*parent_doc[field.solr_name])
-          
-        #if child value is present and no ADD inheritance, skip
-        elsif solr_doc[field.solr_name].present?
-          next
-        #if no child value present, inherit from parent
-        else
-          solr_doc[field.solr_name] = parent_doc[field.solr_name]
```
Comment (rschwab) [inline app/indexers/work_indexer.rb:117]: Don't we still need this for non-ADD inheritance?

File: app/indexers/work_indexer.rb [hunk 2]
```diff
@@ -130,14 +130,15 @@ def inherit_fields_from_parent_collection solr_doc
...
-        #if no child value present, inherit from parent collection
-        else
-          solr_doc[field.solr_name] = parent_doc[field.solr_name]
+        #commenting the below code and not removing it since we might need it for future use. 
+        # #if child value is present and no ADD inheritance, skip
+        # elsif solr_doc[field.solr_name].present?
+        #   next
+        # #if no child value present, inherit from parent collection
+        # else
+        #   solr_doc[field.solr_name] = parent_doc[field.solr_name]
```
Comment (rschwab) [inline app/indexers/work_indexer.rb:141]: This code looks like its an exact copy, a good candidate for its own function (ie inherit_field)

**Review by rschwab — COMMENTED** [review]
It seems like this removes the ability for inheriting the fields that aren't ADD inherited fields. Am I wrong about that?

File: app/indexers/work_indexer.rb (hunk 1 above)
Comment (snehagunduraoUL) [reply app/indexers/work_indexer.rb:117]: Not sure what has changed, but hyrax is using "inherit_metadata" in app/models/work.rb for non-ADD inheritance. The solr document is updated with child values in work_indexer.rb already.

Comment (snehagunduraoUL) [reply app/indexers/work_indexer.rb:117]: Also, this comment mentions the same. https://github.com/UCSCLibrary/ucsc-library-digital-collections/blob/master/app/indexers/work_indexer.rb#L93

Comment (rschwab) [reply app/indexers/work_indexer.rb:117]: I see this comment:
```
  # Metadata inheritance can also happen when the work is saved, in which case the
  # inherited metadata is stored in Fedora and indexed normally.
  # This method comes in to play when you want the inherited metadata to display for users
  # but not to be part of the preservation record.
```
which explains why its in two places. 

So I guess non-add inheritance is saved to the fedora record, and add inheritance just to the solr index (ie searchable and on the website). I wonder if that is intentional? @rmjaffe ?

File: app/indexers/work_indexer.rb (hunk 2 above)
Comment (snehagunduraoUL) [reply app/indexers/work_indexer.rb]: Good point. Done.

File: app/indexers/work_indexer.rb (hunk 1 above)
Comment (rmjaffe) [reply app/indexers/work_indexer.rb:117]: Eeps, I have no idea.  Is there be any need or advantage to treating the add inheritance differently than the other inherited values?

Comment (snehagunduraoUL) [reply app/indexers/work_indexer.rb:117]: ADD-inheritance did not exist previously. We can save that to fedora record if we need to. 
I can explain this over a call on Tuesday.

Comment (rschwab) [reply app/indexers/work_indexer.rb:117]: Cool let me know what you come up with - consider this PR approved pending this decision.

**Pull request closed without merging**